### PR TITLE
DAH-1137 Right panel app, what to expect, move in, help sections

### PR DESCRIPTION
**Pull request by emilyjablonski**
[DAH-1137](https://sfgovdt.jira.com/jira/software/c/projects/DAH/boards/122?modal=detail&selectedIssue=DAH-1137)

This ticket/PR adds the how to apply, get application, submit application, need help, and what to expect sections into the sidebar.

It takes advantage of some new components in Bloom, namely the `SidebarBlock` component, so that we combine Bloom components piecemeal to build up the sidebar, rather than consuming the `GetApplication` component in Bloom which while it looks extremely similar, just has a few functional / toggleable differences.

It also pulls the waitlist block into its own file, since it had different toggleable logic from the entire apply section and had gotten a little hefty, and moves out the What to Expect since that always renders.

Some different states:
Rentals:
Housing Counselor link goes to internal `housing-counselors`, and it includes some helper text
listings/a0W4U00000JqTLIUA3 Open applications without paper apps
listings/a0W0P00000GwGl3UAF Open applications with paper apps
listings/a0W4U00000KnHmaUAF Upcoming lottery, should not render
listings/a0W4U00000KnHO6UAN Lottery results, should not render

Sale:
Housing Counselor link goes to external URL, and it does not have the helper text
How to Apply section has some eligibility helper text
Has Expected Move In Date section
listings/a0W0P00000GlKfBUAV Open applications without paper apps
listings/a0W0P00000HbqLvUAJ Open applications with paper apps
listings/a0W4U00000KnHu5UAF Upcoming lottery, should not render
listings/a0W4U00000KnHgrUAF?react=false Lottery results, should not render

### CODE REVIEW

File: app/javascript/modules/listingDetailsAside/ListingDetailsApply.tsx [hunk 1]
```diff
@@ -1,85 +1,164 @@
...
+                url: "https://sfmohcd.org/homebuyer-program-eligibility",
+              })
+            )}
+          </p>
+          <p className={"mb-4"}>{t("listings.apply.eligibilityRequirementDescription")}</p>
+        </>
+      )}
+      <Button styleType={AppearanceStyleType.primary} className={"w-full"} transition={true}>
+        {t("label.applyOnline")}
+      </Button>
```
[the remark this answers is not in the document]
Comment (emilyjablonski) [reply app/javascript/modules/listingDetailsAside/ListingDetailsApply.tsx]: Ah yes that's important isn't it lol 🙃 Fixed!

**Comment by christine-sfg** [comment]
Thanks for all the example listings! Super helpful for testing.

I added some testing notes to [this doc](https://docs.google.com/document/d/1Ir0IvhqzMINmwsFd4LMxY8OSdO-oyHpl5KYU-OafK8w/edit?usp=sharing)

File: app/javascript/util/listingUtil.ts [hunk 2]
```diff
@@ -72,3 +73,53 @@ export const getListingAddressString = (listing: RailsListing): string => {
...
+/**
+ * Check if a listing is accepting paper applications
+ * @param {RailsRentalListing | RailsRentalListing} listing
+ * @returns {boolean} returns true if the listing is accepting paper applications, false otherwise
+ */
+export const acceptingPaperApplications = (listing: RailsListing): boolean => {
+  return listing.Accepting_applications_at_leasing_agent || listing.Accepting_applications_by_PO_Box
+}
+
+export const mohcdPaperAppURLBase = "https://sfmohcd.org/sites/default/files/Documents/MOH/"
```
Comment (christine-sfg) [inline app/javascript/util/listingUtil.ts:86]: do these 3 vars need to be exported? perhaps move them into the `paperApplicationURLs` function?

File: app/javascript/util/listingUtil.ts [hunk 3]
```diff
@@ -72,3 +73,53 @@ export const getListingAddressString = (listing: RailsListing): string => {
...
+ * @returns {PaperApplication[]}
+ */
+export const paperApplicationURLs = (isRental: boolean): PaperApplication[] => {
+  const urlBase = isRental ? mohcdRentalPaperAppURLTemplate : mohcdSalePaperAppURLTemplate
+  return paperAppLanguages.map(
+    (lang): PaperApplication => {
+      return {
+        languageString: LANGUAGE_CONFIGS[lang.prefix].getLabel(),
+        fileURL: urlBase.replace("{lang}", lang.language),
+      }
```
Comment (christine-sfg) [inline app/javascript/util/listingUtil.ts:122]: unit test for this function would be cool to make sure we're building all the url's properly

**Review by christine-sfg — COMMENTED** [review]
Great work! Just a few comments in addition to the testing feedback doc.

**Comment by emilyjablonski** [comment]
@christine-sfg Made all the updates! 😁

File: app/javascript/modules/listingDetailsAside/ListingDetailsApply.tsx [hunk 4]
```diff
@@ -1,85 +1,183 @@
...
+          <Button
+            className={"w-full mt-4"}
+            transition={true}
+            icon={"arrowDown"}
+            iconPlacement={"right"}
+            onClick={() => setPaperApplicationsOpen(!paperApplicationsOpen)}
+          >
+            {t("label.downloadApplication", {
+              lang: "",
+            })}
```
Comment (christine-sfg) [inline app/javascript/modules/listingDetailsAside/ListingDetailsApply.tsx:90]: could we remove this lang param from here and the text props?

File: app/javascript/modules/listingDetailsAside/ListingDetailsApply.tsx [hunk 5]
```diff
@@ -1,85 +1,183 @@
...
+        title={t("listings.apply.sendByUsMail")}
+      >
+        <div className={"mb-2 text-gray-900 text-base"}>
+          <MultiLineAddress
+            address={{
+              placeName: listing.Application_Organization,
+              street: listing.Application_Street_Address,
+              city: listing.Application_City,
+              zipCode: listing.Application_Postal_Code,
+              state: listing.Application_State,
```
Comment (christine-sfg) [inline app/javascript/modules/listingDetailsAside/ListingDetailsApply.tsx:133]: For accessibility, we may want to update this bloom component to use similar logic added for [DAH-50](https://sfgovdt.jira.com/browse/DAH-50)

File: app/javascript/modules/listingDetailsAside/ListingDetailsApply.tsx [hunk 6]
```diff
@@ -1,85 +1,183 @@
...
+            }}
+          />
+        </div>
+        {t("listings.apply.applicationsMustBeReceivedByDeadline")}
+      </SidebarBlock>
+      <SidebarBlock className={"bg-blue-200"}>
+        {`${t("listings.apply.doNotApplyOnlineAndPaper")} ${t(
+          "listings.apply.doNotDoMultipleApplications"
+        )}`}
```
Comment (christine-sfg) [inline app/javascript/modules/listingDetailsAside/ListingDetailsApply.tsx:142]: hmm, concatenating strings is not a good idea. i think we should consolidate these two into one string and have it re-translated.

**Review by christine-sfg — COMMENTED** [review]
Good job on all the fixes! Looks great! Just a couple more comments.

File: app/javascript/modules/listingDetailsAside/ListingDetailsWaitlist.tsx [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+import { QuantityRowSection, t } from "@bloom-housing/ui-components"
+import { RailsListing } from "../listings/SharedHelpers"
+import { isLotteryComplete, isOpen } from "../../util/listingUtil"
+
+export interface ListingDetailsWaitlistProps {
+  listing: RailsListing
+}
+
+export const ListingDetailsWaitlist = ({ listing }: ListingDetailsWaitlistProps) => {
+  if (!listing.hasWaitlist || (!isOpen(listing) && !isLotteryComplete(listing))) return null
```
Comment (kramduckner) [inline app/javascript/modules/listingDetailsAside/ListingDetailsWaitlist.tsx:11]: any preference or thoughts on returning null vs <></>?

**Comment by kramduckner** [comment]
Looks great! In the review instructions, the text says 'should not render' on a few listings - what shouldn't be rendering?

File: app/javascript/modules/listingDetailsAside/ListingDetailsWaitlist.tsx (hunk 7 above)
Comment (christine-sfg) [reply app/javascript/modules/listingDetailsAside/ListingDetailsWaitlist.tsx:11]: @kramduckner good question. the react docs recommend returning null https://reactjs.org/docs/conditional-rendering.html#preventing-component-from-rendering

File: app/javascript/modules/listingDetailsAside/ListingDetailsApply.tsx (hunk 4 above)
Comment (emilyjablonski) [reply app/javascript/modules/listingDetailsAside/ListingDetailsApply.tsx:90]: Can do! Needed to add a new string

File: app/javascript/modules/listingDetailsAside/ListingDetailsApply.tsx (hunk 5 above)
Comment (emilyjablonski) [reply app/javascript/modules/listingDetailsAside/ListingDetailsApply.tsx:133]: Absolutely I'll make a Bloom ticket

File: app/javascript/modules/listingDetailsAside/ListingDetailsApply.tsx (hunk 6 above)
Comment (emilyjablonski) [reply app/javascript/modules/listingDetailsAside/ListingDetailsApply.tsx:142]: Added a new string!

**Comment by emilyjablonski** [comment]
@kramduckner Good question! None of the new components should render in the case of an open lottery or lottery results bc there are no longer applications available at that stage

**Review by christine-sfg — APPROVED** [review]
lgtm! great job!

**Pull request merged**